### ISSUE TITLE
Fix type field

### ISSUE THREAD

**Issue by dbrgn** [coredump-ch/spaceapi#2]
Apparently we can't define a field in a struct named `type` because that's a reserved keyword.

After a quick question in IRC:

```irc
18:45 < dbrgn> hi. how do I serialize a field named "type" to JSON? "type" is a reserveed keyword, therefore it cannot be used in a struct.
18:50 < mbrubeck> dbrgn: You could use a HashMap instead of a struct, but of course that loses a lot of static checking...
22:27 < flan3002> dbrgn - It's nasty, but you could certainly alter the derived implementation of rustc_serialize::Encode.
22:28 < mbrubeck> dbrgn: There's already some automatic mapping for things like hyphens/underscores IIRC, so that seems like a decent way to go... Not sure how to implement it though.
```

**Comment by rnestler** [comment]
Almost done, will write a new blogpost about the solution ;)

**Comment by dbrgn** [comment]
Yaay! :)

**Comment by dbrgn** [comment]
(But why on the Redis branch? :))

**Comment by rnestler** [comment]
@dbrgn 
Because it was late in the evening and I didn't notice I was working on the Redis branch :wink:
I could do some ``git rebase`` magic if you want...

**Comment by dbrgn** [comment]
Yeah, why not. I'll probably continue work on the redis branch soon.

**Comment by rnestler** [comment]
So I did some rebase magic. This will probably break your local redis branch..

**Comment by dbrgn** [comment]
No problem!

**Comment by dbrgn** [comment]
Fixed with 60612bdb833b166ad2ef187bf8c288b5e84be1b3.
